src/cli/README_DATASET.md

```diff
@@ -107,6 +107,108 @@ After the device successfully attaches to a Thread network, the device will retr
    Done
    ```
 
+### Using the Pending Operational Dataset for Delayed Dataset Updates
```

Comment:
As an FYI we have a concepts doc based on this one that will need an update too.  https://github.com/openthread/ot-docs/blob/main/site/en/reference/cli/concepts/dataset.md

That one gets hosted on openthread.io: https://openthread.io/reference/cli/concepts/dataset

I can have someone update that later once this is in.